### PR TITLE
Refactoring part of Incomes Matching PR

### PR DESCRIPTION
**Pull request by jiivan**
Some refactorings that prepare ground for Incomes Matching.

### CODE REVIEW

File: golem/network/transport/message.py [hunk 1]
```diff
@@ -179,6 +179,24 @@ def __repr__(self):
         return "{}".format(self.__class__)
 
 
+class MessageRedefined(Message):
```
Comment (badb) [inline golem/network/transport/message.py]: This needs another name. Like MessageWithMapping. Or just Message and we can replace all other messages with this class.

File: golem/task/tasksession.py [hunk 2]
```diff
@@ -38,6 +33,17 @@ def call_task_computer_and_drop_after_attr_error(*args, **kwargs):
...
+        @functools.wraps(f)
+        def curry(self, *args, **kwargs):
+            result = f(self, *args, **kwargs)
+            self.dropped()
+            return result
+        return curry
+    return inner
+
+
 class TaskSession(MiddlemanSafeSession):
```
Comment (badb) [inline golem/task/tasksession.py:47]: We should increase protocol version here (if we're introducing new type of Messages)

File: golem/task/tasksession.py [hunk 3]
```diff
@@ -278,34 +277,25 @@ def send_report_computed_task(self, task_result, address, port, eth_account, nod
             return
         node_name = self.task_server.get_node_name()
 
-        self.send(MessageReportComputedTask(task_result.subtask_id, task_result.result_type, task_result.computing_time,
-                                            node_name, address, port, self.task_server.get_key_id(), node_info,
-                                            eth_account, extra_data))
+        self.send(message.MessageReportComputedTask(task_result.subtask_id, task_result.result_type, task_result.computing_time, node_name, address, port, self.task_server.get_key_id(), node_info, eth_account, extra_data))
```
Comment (badb) [inline golem/task/tasksession.py:280]: What line length do you suggest using? Previously we've used 120, right now I'm trying to use 100, but it was never directly specified.

**Review by badb — CHANGES_REQUESTED** [review]
Very cool, but needs more unittests, especially new Message class and tasksession/taskserver refactor functions.

File: golem/task/taskserver.py [hunk 4]
```diff
@@ -518,9 +528,9 @@ def __connection_for_task_request_failure(self, conn_id, node_name, key_id, task
             pc.status = PenConnStatus.WaitingAlt
             pc.time = time.time()
 
-    def __connection_for_task_result_established(self, session, conn_id, key_id, waiting_task_result):
-        self.remove_forwarded_session_request(key_id)
-        session.key_id = key_id
+    def __connection_for_task_result_established(self, session, conn_id, waiting_task_result):
+        self.remove_forwarded_session_request(waiting_task_result.owner_key_id)
```
Comment (Elfoniok) [inline golem/task/taskserver.py:532]: Are you sure that key_id is always owner_key_id?

File: golem/task/taskserver.py [hunk 5]
```diff
@@ -532,18 +542,17 @@ def __connection_for_task_result_established(self, session, conn_id, key_id, wai
...
 
-    def __connection_for_task_result_failure(self, conn_id, key_id, waiting_task_result):
+    def __connection_for_task_result_failure(self, conn_id, waiting_task_result):
 
         def response(session):
-            self.__connection_for_task_result_established(
-                session, conn_id, key_id, waiting_task_result)
+            self.__connection_for_task_result_established(session, conn_id, waiting_task_result)
 
         if key_id in self.response_list:
```
Comment (Elfoniok) [inline golem/task/taskserver.py]: key_id not defined here?

File: golem/task/taskserver.py [hunk 6]
```diff
@@ -532,18 +542,17 @@ def __connection_for_task_result_established(self, session, conn_id, key_id, wai
...
             self.response_list[conn_id].append(response)
         else:
             self.response_list[conn_id] = deque([response])
 
-        self.client.want_to_start_task_session(key_id, self.node, conn_id)
+        self.client.want_to_start_task_session(wating_task_result, key_id, self.node, conn_id)
```
Comment (Elfoniok) [inline golem/task/taskserver.py]: key_id not defined here? ...."wating" there is no such thing! One more thing you changed passed arguments list, but i don't see modification of Client function definition

File: golem/task/tasksession.py (hunk 2 above)
Comment (jiivan) [reply golem/task/tasksession.py:47]: Actually new messages will be introduced in second PR. Here the refactoring doesn't change the external representation of Msg.

**Review by Elfoniok — CHANGES_REQUESTED** [review]
There are errors to fix

File: golem/task/tasksession.py (hunk 3 above)
Comment (jiivan) [reply golem/task/tasksession.py:280]: [PEP8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) recommends something between 72 and 100 characters. I personally disagree with those limits. I feel that all moderately modern workstations are able to properly display any line length (with or without wrapping). That's why I disabled line length checks in [style checker](https://github.com/golemfactory/golem/blob/develop/golem/testutils.py#L137).

And this diff shows why I feel single lines are easier to maintain. When I added `message` namespace I just didn't want to go ascii arting with [aligned indentation](https://www.python.org/dev/peps/pep-0008/#indentation). That's why I quickly joined all lines. This won't be a problem with simple hanging indents though.

But I'm flexible in this matter. If most of you agree with line length restrictions I will enable them.

File: golem/task/taskserver.py (hunk 4 above)
Comment (jiivan) [reply golem/task/taskserver.py:532]: Yes. SEE: https://github.com/golemfactory/golem/pull/1026/files/47a21c8720eecc0be34a7f0e876e273715d67951#diff-58d817e424421f899f53335aa2effd0eL797

Comment (Elfoniok) [reply golem/task/taskserver.py:532]: What about call site present inside __connection_for_task_result_failure ?

Comment (jiivan) [reply golem/task/taskserver.py:532]: https://github.com/golemfactory/golem/pull/1026/files/47a21c8720eecc0be34a7f0e876e273715d67951#diff-58d817e424421f899f53335aa2effd0eL535

Comment (Elfoniok) [reply golem/task/taskserver.py:532]: OK

Comment (Elfoniok) [reply golem/task/taskserver.py:532]: OK

File: tests/golem/task/test_taskserver.py [hunk 7]
```diff
@@ -520,6 +521,35 @@ def test_conn_final_failures(self):
         self.assertTrue(ts.remove_responses.called)
         self.assertTrue(ts.task_computer.session_timeout.called)
 
+    def test_task_result_connection_failure(self):
```
Comment (jiivan) [inline tests/golem/task/test_taskserver.py:524]: @Radagast-red ok?

Comment (Elfoniok) [reply tests/golem/task/test_taskserver.py:524]: Ok

**Review by Elfoniok — APPROVED** [review]
As a general note, i understand that you have right in general, But as at least some tools (github for example) poorly support lines longer than 120 columns. I would opt for keeping and formatting code according to this.

**Review by badb — APPROVED** [review]

File: golem/model.py [hunk 8]
```diff
@@ -138,6 +142,9 @@ def __init__(self, *args, **kwargs):
         if self.details is None:
             self.details = {}
 
+    def __repr__(self):
+        tx = self.details.get('tx', 'NULL')
+        return "<Payment stid: %r v: %.3f s: %r tx: %s>" % (self.subtask, self.value/denoms.ether, self.status, tx)
```
Comment (mfranciszkiewicz) [inline golem/model.py]: Maybe it's not a big deal, but `%` formatting is inconsistent with the rest of the project.

**Pull request merged**